### PR TITLE
Test CalciteExecutor doing  query in single table.

### PR DESCRIPTION
**Pull request by lmhmhl**
Seeing  issue  #8883

### CODE REVIEW

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java [hunk 1]
```diff
@@ -385,7 +390,11 @@ protected Boolean getSaneResult(final SQLStatement sqlStatement) {
         };
         return driverJDBCExecutor.execute(executionGroups, sqlStatement, routeUnits, jdbcExecutorCallback);
     }
-    
+
+    private void isToCalcite(final boolean isToCalcite) {
```
Comment (tristaZero) [inline shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java:394]: Hi, we can not add `set` or `get` functions for tests. I guess you want to make this property `true`. That way, please refer to the following steps,
- Add another single table to `config-calcite.yaml`
- Modify your test SQL as `select xx,xx from singleTable1, singleTable2 where singleTable1.xx=singleTable2 and xx`
- If necessary, you can add SQLS to prepare the Calcite UT in `jdbc_data.sql` and `jdbc_init.sql`. It is straightforward to create tables to do calcite tests, isn't it?

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/java/org/apache/shardingsphere/driver/jdbc/core/statement/CalcitePrepareStatement.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public final class CalcitePrepareStatement extends AbstractShardingSphereDataSourceForCalciteTest {
```
Comment (tristaZero) [inline shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/java/org/apache/shardingsphere/driver/jdbc/core/statement/CalcitePrepareStatement.java:32]: Hi `CalcitePrepareStatementTest` is a good name, isn't it?

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/java/org/apache/shardingsphere/driver/jdbc/core/statement/CalcitePrepareStatement.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    private static final String INSERT_SQL = "INSERT INTO t_encrypt (id, cipher_pwd, plain_pwd) VALUES (?, ?, ?)";
+
+    private static final String SELECT_SQL_BY_ID = "SELECT id, cipher_pwd, plain_pwd FROM t_encrypt WHERE id = ?";
+
+    @Before
+    public void init() throws SQLException {
+        try (PreparedStatement statement = getShardingSphereDataSource().getConnection().prepareStatement(INSERT_SQL)) {
```
Comment (tristaZero) [inline shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/java/org/apache/shardingsphere/driver/jdbc/core/statement/CalcitePrepareStatement.java:40]: Hi, we can consider inserting some data in `jdbc_data.sql` which is designed for initialization.

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/resources/config-calcite.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+rules:
+  - !SHARDING
+    tables:
+      t_encrypt:
+        actualDataNodes: jdbc_0.t_encrypt
```
Comment (tristaZero) [inline shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/resources/config-calcite.yaml:22]: Hi, could add a new line below this line?

**Review by tristaZero — CHANGES_REQUESTED** [review]
Hi @lmhmhl,

Considering this is a `real` PR from you, It looks nice than expected.
Still, I wrote some comments for your reference.
Good luck!

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java (hunk 1 above)
Comment (lmhmhl) [reply shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java:394]: > Hi, we can not add `set` or `get` functions for tests. I guess you want to make this property `true`. That way, please refer to the following steps,
> 
> * Add another single table to `config-calcite.yaml`
> * Modify your test SQL as `select xx,xx from singleTable1, singleTable2 where singleTable1.xx=singleTable2 and xx`
> * If necessary, you can add SQLS to prepare the Calcite UT in `jdbc_data.sql` and `jdbc_init.sql`. It is straightforward to create tables to do calcite tests, isn't it?

Thanks for your encouragement, you are so kind.  I have a  questions, how to go the branch as follows when test？It seems nowhere can set the boolean value `toCalcite` in `RouteContext`
```
 if (executionContext.getRouteContext().isToCalcite()) {
            return executeQueryByCalcite();
        }
```

Comment (tristaZero) [reply shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java:394]: Hi, please look at Line 57 in `SingleTableRoutingEngine.java`

Comment (tristaZero) [reply shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java:394]: That's why I suggest you add another single table to config-calcite.yaml and modify your test SQL.

**Comment by lmhmhl** [comment]
> Hi @lmhmhl,
> 
> Considering this is a `real` PR from you, It looks nice than expected.
> Still, I wrote some comments for your reference.
> Good luck!

Hi. @tristaZero
I have modified the code followed by your suggestions, and add a single table from another database to do query from different database instances. Please review my latest modified code. Thx.

**Review by tristaZero — APPROVED** [review]
Hi @lmhmhl,

I have to say your PR is so great than expected! 
It is not just a unit test but rather a `test engine` for later calcite tests. Please consider staying in this community. I really want to see more of your interaction in the future if you also enjoy here. :)

Yours,
Trista

**Comment by lmhmhl** [comment]
> Hi @lmhmhl,
> 
> I have to say your PR is so great than expected!
> It is not just a unit test but rather a `test engine` for later calcite tests. Please consider staying in this community. I really want to see more of your interaction in the future if you also enjoy here. :)
> 
> Yours,
> Trista
Hi @tristaZero 
Thanks for your compliment, I wanna do the other related calcite test in the future. I love challenges and hope to play a more important role in the future. You can dispatch me some extra tasks. I'd love to do them.

**Comment by tristaZero** [comment]
Hi @lmhmhl ,

Great, I will prepare the next issues for you these days. Please be patient for my comment here. :)

**Pull request merged**